projects/gloo/pkg/upstreams/consul/consul_client.go

```diff
@@ -91,6 +128,27 @@ func (c *consul) filter(dataCenters []string) []string {
 	return filtered
 }
 
+// Filters out the services that do not have matching tags from the service_tags_allowlist
+//input from services is a map of service name to slice of tags
+func (c *consul) filterServices(services map[string][]string) map[string][]string {
+	filteredServices := make(map[string][]string)
+	for serviceName, sTags := range services {
+
+		//if there is no allowlist, allow for all services
+		if len(c.serviceTagsAllowlist) == 0 {
+			return services
+		}
+		//Filter services by tags
+		for _, tag := range sTags {
+			if _, found := c.serviceTagsAllowlist[tag]; found {
+				filteredServices[serviceName] = sTags
+				break
+			}
+		}
+	}
+	return filteredServices
```

Comment:
nit, my suggestion to flatten the code and make it more easily readable was
```suggestion
	//if there is no allowlist, allow for all services
	if len(c.serviceTagsAllowlist) == 0 {
		return services
	}
	filteredServices := make(map[string][]string)
	for serviceName, sTags := range services {
		//Filter services by tags
		for _, tag := range sTags {
			if _, found := c.serviceTagsAllowlist[tag]; found {
				filteredServices[serviceName] = sTags
				break
			}
		}
	}
	return filteredServices
```